packages/core-mobile/app/screens/send/hooks/usePVMSend.ts

```diff
@@ -27,36 +32,89 @@ const usePVMSend: SendAdapterPVM = ({
     amount,
     canValidate
   } = useSendContext()
+  const [gasPrice, setGasPrice] = useState<bigint>()
+  const [feeState, setFeeState] = useState<pvm.FeeState>()
+
+  const [estimatedFee, setEstimatedFee] = useState<bigint>()
+
+  const [provider, setProvider] = useState<Avalanche.JsonRpcProvider>()
+
+  useEffect(() => {
+    const getProvider = async (): Promise<void> => {
+      const p = await NetworkService.getAvalancheProviderXP(
+        !!network.isTestnet,
+        isDevnet(network)
+      )
+      setProvider(p)
+    }
+    getProvider()
+  }, [network])
+
+  useEffect(() => {
+    if (provider && provider.isEtnaEnabled()) {
+      provider
+        .getApiP()
+        .getFeeState()
+        .then(state => {
+          setFeeState(state)
+        })
+        .catch(() => {
+          setError(SendErrorMessage.INVALID_NETWORK_FEE)
+        })
+    } else {
+      setFeeState(undefined)
+    }
+  }, [provider, setError])
+
+  const getFeeState = useCallback(
+    (p?: bigint) => {
+      if (p && feeState) {
+        return {
+          ...feeState,
+          price: p
+        }
+      }
+      if (p === undefined && feeState) {
+        return feeState
+      }
+      return undefined
+    },
+    [feeState]
+  )
+
+  const createSendPTx = useCallback(
+    async (price?: bigint): Promise<UnsignedTx> => {
+      if (!toAddress || !token || !amount) {
+        return Promise.reject('missing required fields')
+      }
```

Comment:
this should live outside as before since it's not really related to the logic in here